### PR TITLE
Fix BC Layer tests

### PR DESCRIPTION
**Pull request by soyuka**

### CODE REVIEW

File: docs/adr/0001-resource-identifiers.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Resource Identifiers
 
-* Status: proposed
+* Status: superseded by [0003-uri-variables](0003-uri-variables.md)]
```
Comment (norkunas) [inline docs/adr/0001-resource-identifiers.md:3]:
```suggestion
* Status: superseded by [0003-uri-variables](0003-uri-variables.md)
```

File: tests/Fixtures/app/AppKernel.php [hunk 2]
```diff
@@ -231,8 +231,6 @@ class_exists(NativePasswordHasher::class) ? 'password_hashers' : 'encoders' => [
             $c->prependExtensionConfig('api_platform', ['enable_nelmio_api_doc' => true]);
         }
 
-        if ($_SERVER['METADATA_BACKWARD_COMPATIBILITY_LAYER'] ?? false) {
-            $c->prependExtensionConfig('api_platform', ['metadata_backward_compatibility_layer' => true]);
-        }
+        $c->prependExtensionConfig('api_platform', ['metadata_backward_compatibility_layer' => ($_SERVER['METADATA_BACKWARD_COMPATIBILITY_LAYER'] ?? false) ? true : false]);
```
Comment (alanpoulain) [inline tests/Fixtures/app/AppKernel.php:234]:
```suggestion
        $c->prependExtensionConfig('api_platform', ['metadata_backward_compatibility_layer' => (bool) ($_SERVER['METADATA_BACKWARD_COMPATIBILITY_LAYER'] ?? false)]);
```

File: docs/adr/0003-uri-variables.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+* Status: accepted
+* Deciders: @dunglas @soyuka
+
+Implementation: [#4408][pull/4408]
+
+## Context and Problem Statement
+
+For reference see the previously implemented [Resource identifier](0001-resource-identifiers) ADR. 
+URI variables are the URI template (e.g. `/books/{id}`) variables (e.g. `id`). When defining alternate resources (see [Resource definition](0002-resource-definition)), we need more informations then only a property and a class. The tuple accepted in [Resource identifier](0001-resource-identifiers) is not enough. For example:
```
Comment (alanpoulain) [inline docs/adr/0003-uri-variables.md:11]:
```suggestion
URI variables are the URI template (e.g. `/books/{id}`) variables (e.g. `id`). When defining alternate resources (see [Resource definition](0002-resource-definition)), we need more information than only a property and a class. The tuple accepted in [Resource identifier](0001-resource-identifiers) is not enough. For example:
```

Comment (soyuka) [reply docs/adr/0003-uri-variables.md:11]: thx fixed in https://github.com/api-platform/core/pull/4441

File: docs/adr/0001-resource-identifiers.md (hunk 1 above)
Comment (soyuka) [reply docs/adr/0001-resource-identifiers.md:3]: thx fixed in https://github.com/api-platform/core/pull/4441

**Pull request closed without merging**